src/ui/nFormat.ts

```diff
@@ -87,6 +87,10 @@ export function nFormatNew(n: number, options: NFormatOptions = {}) {
   if (!suffixList[suffixIndex]) return exponentialFormatter.format(n).toLocaleLowerCase();
   // Suffixed form
   n /= logBase ** suffixIndex;
+  if (Math.abs(n) >= logBase - 1 + parseFloat("0." + "9".repeat(fractionalDigits)) && suffixList[suffixIndex + 1]) {
```

Comment:
This probably deserves a comment, since I'm looking at both the code and test together, and I'm still not sure why you had to add it.

Reply:
Yeah this unfortunately was the best way I could think of to force it to use the next highest suffix if it was going to round to something that shouldn't display (e.g. 1,000.000m instead of 1.000b)

Reply:
I added a comment but I'm not marking this as resolved yet as I'd like to improve this before merging.